third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -767,17 +775,39 @@ impl Interpreter {
     // Debugging and logging helpers.
     //
 
-    /// Given an `VMStatus` generate a core dump if the error is an `InvariantViolation`.
-    fn maybe_core_dump(&self, mut err: VMError, current_frame: &Frame) -> VMError {
-        // a verification error cannot happen at runtime so change it into an invariant violation.
+    /// If the error is invariant violation, attaches the state of the current frame.
+    fn attach_state_if_invariant_violation(
+        &self,
+        mut err: VMError,
+        current_frame: &Frame,
+    ) -> VMError {
+        // A verification error can be returned when
+        //   1) some check fails at runtime, e.g. type layout has too many type
+        //      nodes,
+        //   2) bytecode verifier fails, e.g. on module publishing.
+        // These errors mean that the code breaks some invariant, so we need to
+        // remap the error to VERIFICATION_ERROR which is of invariant violation
+        // type.
         if err.status_type() == StatusType::Verification {
```

Comment:
This seems kind of error-prone, as you're manually copying everything except the `StatusCode` from a `MVError` to a new one.  Why not a new function to `VMError` to replace the major status?  This outsourcing of functionality is probably why some is missing today.

Reply:
Good point, updated